addons/hr_expense_stripe/utils.py

```diff
@@ -470,54 +451,54 @@ def format_amount_to_stripe(amount, currency):
+            # Add sub dict to this dict
+            for sub_dict_key, sub_dict_item in new_sub_dict.items():
+                new_dict[f'{dict_key}{sub_dict_key}'] = sub_dict_item
+        else:
+            new_dict[dict_key] = item
+    return new_dict
+
+def make_request_stripe_proxy(route, payload=None, method="POST", headers=None, test_mode=False):
+    if not headers:
+        headers = {}
 
-    url = '/'.join(('https://api.stripe.com/v1', endpoint))
     headers = {
-        'AUTHORIZATION': f'Bearer {api_key}',
-        'Stripe-Version': API_VERSION,  # SetupIntent requires a specific version.
+        'Content-Type': 'application/json',
+        **headers,
     }
-    if method == 'POST' and idempotency_key:
-        headers['Idempotency-Key'] = idempotency_key
+
+    proxy_server = 'https://stripe-issuing-proxy.test.odoo.com' if test_mode else 'https://stripe-issuing-proxy.api.odoo.com'
+    proxy_server = "http://127.0.0.1:8070"  # Temp local IAP server
+    url = f"{proxy_server}/{route}"
     try:
-        response = requests.request(method, url, data=payload, headers=headers, timeout=60)
-        # Stripe can send 4XX errors for payment failures (not only for badly-formed requests).
-        # Check if an error code is present in the response content and raise only if not.
-        # See https://stripe.com/docs/error-codes.
-        # If the request originates from an offline operation, don't raise to avoid a cursor
-        # rollback and return the response as-is for flow-specific handling.
-        if (
-                not response.ok
-                and not offline
-                and 400 <= response.status_code < 500
-                and response.json().get('error')
-        ):  # The 'code' entry is sometimes missing
-            try:
-                response.raise_for_status()
-            except requests.exceptions.HTTPError:
-                _logger.exception("invalid API request at %s with data %s", url, payload)
-                error_msg = response.json().get('error', {}).get('message', '')
-                raise ValidationError(
-                    "Stripe: " + _(
-                        "The communication with the API failed.\n"
-                        "Stripe gave us the following info about the problem:\n'%s'", error_msg
-                    )
-                )
+        response = requests.request(method, url, data=json.dumps(flatten_dict(payload)), headers=headers, timeout=60)
+
+        response_content = response.json()
+        if 'error' in response_content:
+            raise ValidationError(response_content['error']['data']['message'])
     except requests.exceptions.ConnectionError:
         _logger.exception("unable to reach endpoint at %s", url)
-        raise ValidationError("Stripe: " + _("Could not establish the connection to the API."))
-    return response.json()
+        raise ValidationError("Stripe Proxy: " + _("Could not establish the connection to the Proxy Server."))
```

Comment:
```suggestion
        raise ValidationError(_("Stripe Proxy: Could not establish the connection to the Proxy Server."))
```
Don't forget RTL :p

Reply:
TBH it's taken from payment and they're doing the same, and I didn't check when copy-pasting :sweat_smile: